### PR TITLE
Add separate section to bulk move by custom taxonomy

### PR DESCRIPTION
**Pull request by mariadanieldeepak**
The following are the tasks done as part of this task.

- Reviewed the [PR](https://github.com/sudar/bulk-move/pull/16)
- Review comments that you mentioned have been implemented

### CODE REVIEW

**Comment by mariadanieldeepak** [comment]
@sudar,

Review comments have been fixed. Also, the Taxonomy layer has been added.
https://github.com/sudar/bulk-move/tree/dev/1.3

File: bulk-move.php [hunk 1]
```diff
@@ -188,6 +191,8 @@ private function setup_actions() {
         // Register hooks
         add_action( 'admin_menu', array( &$this, 'add_menu' ) );
         add_action( 'admin_init', array( &$this, 'request_handler' ) );
+        add_action( 'wp_ajax_load_custom_taxonomy_by_post_type', 'Bulk_Move_Posts::load_custom_taxonomy_by_post_type' );
```
Comment (sudar) [inline bulk-move.php]: Any specific reason for making these as static methods?

File: include/class-bulk-move-posts.php [hunk 2]
```diff
@@ -446,6 +446,215 @@ public static function change_timeout() {
...
+
+		$args = array(
+			'taxonomy'   => $taxonomies,
+			'hide_empty' => false,
+			'orderby'    => 'name',
+		);
+
+		ob_start();
+        ?>
+		<option class="level-0" value="select">Select Taxonomy&nbsp;&nbsp;</option>
```
Comment (sudar) [inline include/class-bulk-move-posts.php]: Make this text available for translation

File: bulk-move.php (hunk 1 above)
Comment (mariadanieldeepak) [reply bulk-move.php]: All the methods in the `Bulk_Move_Posts` is static. To make it easier when testing, I stayed consistent with other methods.

File: include/class-bulk-move-posts.php (hunk 2 above)
Comment (mariadanieldeepak) [reply include/class-bulk-move-posts.php]: Perfect catch. I made the string translatable and also made the other strings translatable.

File: include/class-bulk-move-posts.php [hunk 3]
```diff
@@ -446,6 +446,215 @@ public static function change_timeout() {
...
+					</td>
+					<td scope="row">
+				</tr>
+				<tr>
+					<td scope="row" colspan="2">
+					<?php
+						$custom_post_types_args = array( '_builtin' => false );
+						$custom_post_types      = get_post_types( $custom_post_types_args );
+					?>
+						<?php // TODO: Hide <select> tags and show messaged when no custom post types are available. ?>
```
Comment (sudar) [inline include/class-bulk-move-posts.php]: Let's handle this todo as well.

**Comment by sudar** [comment]
@mariadanieldeepak 

In what branch are you making the changes for this PR? Is it `dev/1.3`?

**Comment by mariadanieldeepak** [comment]
> Let's handle this todo as well.

This has been taken care of.

image

> In what branch are you making the changes for this PR? Is it dev/1.3?

Yes.

**Comment by sudar** [comment]
>> In what branch are you making the changes for this PR? Is it dev/1.3?

> Yes.

In future, let's create seperate branches off the dev/* branches so that we can individually review each branch and merge them.

**Pull request merged**